internal/e2e/transaction_cases_test.go

```diff
@@ -58,6 +60,53 @@ func runTransactionCases(c transactClient, m *namespaceTestManager) func(*testin
 			assert.Len(t, resp.RelationTuples, 0)
 		})
 
+		t.Run("case=large inserts and deletes", func(t *testing.T) {
+			ns := []*namespace.Namespace{
+				{Name: t.Name() + "1"},
+				{Name: t.Name() + "2"},
+			}
+			m.add(t, ns...)
+
+			var tuples []*ketoapi.RelationTuple
+			for i := range 12001 {
+				tuples = append(tuples, &ketoapi.RelationTuple{
+					Namespace: ns[0].Name,
+					Object:    "o" + strconv.Itoa(i),
+					Relation:  "rel",
+					SubjectSet: &ketoapi.SubjectSet{
+						Namespace: ns[1].Name,
+						Object:    "o" + strconv.Itoa(i),
+						Relation:  "rel",
+					},
+				},
+					&ketoapi.RelationTuple{
+						Namespace: ns[0].Name,
+						Object:    "o" + strconv.Itoa(i),
+						Relation:  "rel",
+						SubjectID: pointerx.Ptr("sid"),
+					},
+				)
+			}
+
+			c.transactTuples(t, tuples, nil)
+
+			resp := c.queryTuple(t, &ketoapi.RelationQuery{
+				Namespace: &ns[0].Name,
+			})
+			// Use this instead, but currently unbearably huge output
+			// assert.ElementsMatch(t, tuples, resp.RelationTuples)
+			for i := range tuples {
+				require.Contains(t, resp.RelationTuples, tuples[i])
+			}
```

Comment:
Sorry to break your test here, but it was not properly testing the functionality, as all RTs were the same.
It looks like you have to add pagination here to get the full list of relation tuples now (which you should have done from the beginning).

Reply:
pagination added!